src/ZenohUTransport.cpp

```diff
@@ -103,222 +100,210 @@ ZenohUTransport::uattributesToAttachment(
+v1::UStatus ZenohUTransport::registerResponseListener_(
     const std::string& zenoh_key, CallableConn listener) {
 	std::unique_lock<std::mutex> lock(rpc_callback_map_mutex_);
 	rpc_callback_map_.insert(std::make_pair(zenoh_key, listener));
 
-	return uprotocol::v1::UStatus();
+	return v1::UStatus();
 }
 
-uprotocol::v1::UStatus ZenohUTransport::registerPublishNotificationListener_(
+v1::UStatus ZenohUTransport::registerPublishNotificationListener_(
     const std::string& zenoh_key, CallableConn listener) {
 	// NOTE: listener is captured by copy here so that it does not go out
 	// of scope when this function returns.
-	auto data_handler = [this, listener](const zenoh::Sample& sample) mutable {
+	auto on_sample = [this, listener](const zenoh::Sample& sample) mutable {
 		listener(sampleToUMessage(sample));
-		// invoke_nonblock_callback(&cb_sender, &listener_cloned, Ok(msg));
 	};
 
-	auto subscriber = zenoh::expect<zenoh::Subscriber>(
-	    session_.declare_subscriber(zenoh_key, std::move(data_handler)));
+	auto on_drop = []() {};
+
+	auto subscriber = session_.declare_subscriber(
+	    zenoh_key, std::move(on_sample), std::move(on_drop));
 	{
 		std::unique_lock<std::mutex> lock(subscriber_map_mutex_);
 		subscriber_map_.emplace(listener, std::move(subscriber));
 	}
-	return uprotocol::v1::UStatus();
+	return v1::UStatus();
 }
 
-uprotocol::v1::UStatus ZenohUTransport::sendRequest_(
-    const std::string& zenoh_key, const std::string& payload,
-    const uprotocol::v1::UAttributes& attributes) {
+v1::UStatus ZenohUTransport::sendRequest_(const std::string& zenoh_key,
+                                          const std::string& payload,
+                                          const v1::UAttributes& attributes) {
 	auto source_str =
-	    uprotocol::datamodel::serializer::uri::AsString().serialize(
-	        attributes.source());
+	    datamodel::serializer::uri::AsString().serialize(attributes.source());
 	CallableConn resp_callback;
 	{
 		std::unique_lock<std::mutex> lock(rpc_callback_map_mutex_);
 
 		if (auto resp_callback_it = rpc_callback_map_.find(source_str);
 		    resp_callback_it == rpc_callback_map_.end()) {
-			return uError(uprotocol::v1::UCode::UNAVAILABLE,
-			              "failed to find UUID");
+			return uError(v1::UCode::UNAVAILABLE, "failed to find UUID");
 		} else {
 			resp_callback = resp_callback_it->second;
 		}
 	}
-	auto on_reply = [&](zenoh::Reply&& reply) {
-		auto result = reply.get();
-
-		if (auto sample = std::get_if<zenoh::Sample>(&result)) {
-			resp_callback(sampleToUMessage(*sample));
-		} else if (auto error = std::get_if<zenoh::ErrorMessage>(&result)) {
+	auto on_reply = [&](const zenoh::Reply& reply) {
+		if (reply.is_ok()) {
+			const auto& sample = reply.get_ok();
+			resp_callback(sampleToUMessage(sample));
+		} else {
 			// TODO: error report
+			// std::cout << "Received an error :" <<
+			// reply.get_err().get_payload().deserialize<std::string>() << "\n";
```

Comment:
I would need to confirm this, but I think that if a (non-OK) status code is set in the response attributes commstatus field, the RpcClient will pass that error on to the outer callback instead of the response.